### PR TITLE
Add BibLatex date formats for parsing to Date.java

### PR DESCRIPTION
**Pull request by GuyPuts**
contributes to #2753

Adds date formats set by BibLatex to the Date.java class. Formats are included in the list that was already present in the class. Tested locally.

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Thanks for your contribution. Please link to the issue with e.g Fixes #.... and also please add tests and a changelog entry

File: src/main/java/org/jabref/model/entry/Date.java [hunk 1]
```diff
@@ -44,7 +44,17 @@ public class Date {
...
+                "uuuu.MM.d",                            // covers 2015.10.15
+                "y-",                                   // covers 2015-
+                "-y",                                   // covers -2015
+                "y-M/y-M",                              // covers 2015-01/2015-02
+                "MMMM y/MMMM y",                        // covers January 2015/February 2015
+                "d MMMM y/d MMMM y",                    // covers 20 January 2015/20 February 2015
+                "y G",                                  // covers 1 BC
+                "y G / y G",                            // covers 30 BC / 5 AD
+                "yyyy G / yyyy G",                      // covers 0030 BC / 0005 AD
+                "yyyy-MM G / yyyy-MM G",                // covers 0030-01 BC / 0005-02 AD
```
Comment (calixtus) [inline src/main/java/org/jabref/model/entry/Date.java:56]: These date ranges are not yet parsed correctly. The static parse method must probably be adapted somehow.

**Review by calixtus — CHANGES_REQUESTED** [review]
Please add a changelog entry and most importantly add tests. Thanks!

File: src/main/java/org/jabref/model/entry/Date.java (hunk 1 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/model/entry/Date.java:56]: Yes, see https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/time/format/DateTimeFormatter.html
and https://github.com/JabRef/jabref/blob/170033a37cd1c6b80ef934428ff1e5140f986cda/src/main/java/org/jabref/model/entry/Date.java#L102-L103

There are already existing unit tests which you need to extend.

**Comment by GuyPuts** [comment]
PR was updated

**Comment by koppor** [comment]
@GuyPuts After pushing, you should check the output of the tests going on at GitHub. I put you a screenshot:

image

It says, checkstyle failed. You can go to the files tab to see the errors https://github.com/JabRef/jabref/pull/9720/files

Alternateively, you can execute checkstyle in your ide as described at s

File: src/test/java/org/jabref/model/entry/DateTest.java [hunk 2]
```diff
@@ -77,6 +78,30 @@ void parseYearRange() {
...
+    void parseYearMonthDayRange2() {
+        Date expectedDataRange = new Date(LocalDate.of(2015, Month.JANUARY, 15), LocalDate.of(2015, Month.FEBRUARY, 25));
+        assertEquals(Optional.of(expectedDataRange), Date.parse("15 January 2015 / 25 February 2015"));
+    }
+
+    @Test
+    void parseYearMonthDayRange3() {
+        Date expectedDataRange = new Date(LocalDate.of(2015, Month.JANUARY, 15), LocalDate.of(2015, Month.FEBRUARY, 25));
+        assertEquals(Optional.of(expectedDataRange), Date.parse("15 January 2015/25 February 2015"));
+    }
```
Comment (koppor) [inline src/test/java/org/jabref/model/entry/DateTest.java:103]: Can this be converted to a parametrized test, too?

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,12 +17,12 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We added "Attach file from URL" to right-click context menu to download and store a file with the reference library. [#9646](https://github.com/JabRef/jabref/issues/9646)
 - We enabled updating an existing entry with data from InspireHEP. [#9351](https://github.com/JabRef/jabref/issues/9351)
 - We added a fetcher for the Bibliotheksverbund Bayern (experimental). [#9641](https://github.com/JabRef/jabref/pull/9641)
+- We added BibLatex date formats for parsing dates.
```
Comment (koppor) [inline CHANGELOG.md:20]: Please link the issue:

```suggestion
- We added BibLaTeX date formats for parsing dates. [#2753](https://github.com/JabRef/jabref/pull/2753)
```

**Review by koppor — CHANGES_REQUESTED** [review]
I think besides the small comments, this is good to go?

File: CHANGELOG.md (hunk 3 above)
Comment (Siedlerchr) [reply CHANGELOG.md:20]: Biblatex has no special casing. It's all lowercase

Comment (koppor) [reply CHANGELOG.md:20]: > Biblatex has no special casing. It's all lowercase

Refs https://github.com/koppor/jabref/issues/434

File: src/main/java/org/jabref/model/entry/Date.java [hunk 4]
```diff
@@ -44,7 +44,17 @@ public class Date {
...
+                "uuuu.MM.d",                            // covers 2015.10.15
+                "y-",                                   // covers 2015-
+                "-y",                                   // covers -2015
+                "d MMMM y/d MMMM y",                    // covers 20 January 2015/20 February 2015
+                "y G",                                  // covers 1 BC
+                "y G / y G",                            // covers 30 BC / 5 AD
+                "yyyy G / yyyy G",                      // covers 0030 BC / 0005 AD
+                "yyyy-MM G / yyyy-MM G",                // covers 0030-01 BC / 0005-02 AD
+                "y?",                                    // covers 2023?
+                "d MMMM y",                                    // covers 20 January 2015
```
Comment (koppor) [inline src/main/java/org/jabref/model/entry/Date.java:56]: The space of the comment is a bit off. I think, the intention is that all comments start at the same column.

**Comment by koppor** [comment]
`DateCheckerTest` complains about failing tests

**Review by koppor — CHANGES_REQUESTED** [review]
Since the tests fail, something should be done to get them running.

**Comment by Siedlerchr** [comment]
Follow up task (n oted in the issue)
        /* TODO: The following date formats do not yet work and need to be created with tests
         *      "u G",                                  // covers 1 BC
         *       "u G / u G",                            // covers 30 BC / 5 AD
         *      "uuuu G / uuuu G",                      // covers 0030 BC / 0005 AD
         *      "uuuu-MM G / uuuu-MM G",                // covers 0030-01 BC / 0005-02 AD
         *      "u'-'",                                 // covers 2015-
         *      "u'?'",                                 // covers 2023?
         */

**Review by koppor — APPROVED** [review]

**Pull request merged**